lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_builtin_method/miq_ae_task_builtin.rb

```diff
@@ -0,0 +1,36 @@
+module MiqAeEngine
+  class MiqAeBuiltinMethod
+    module MiqAeTaskBuiltin
+      extend ActiveSupport::Concern
+
+      module ClassMethods
+        def miq_check_provisioned(obj, inputs = {})
+          root = obj.workspace.root
+          task = root['miq_provision']
+          raise ArgumentError, "miq_provision not specified" unless task
+
+          result = task.statemachine_task_status
+          $miq_ae_logger.info("Builtin ProvisionCheck returned <#{result}> for state <#{task.state}> and status <#{task['status']}>")
```

Comment:
@mkanoor Can we replace "ProvisionCheck" here with "check_provisioned"?

Reply:
@mkanoor Are we going to have one file for all task built in methods?

Reply:
@lfu I think we need 1 file for common functions like tasks (check_provisioned, check_service_provisioned, update_provision_status) They are all reading/updating the MiqRequestTasks